### PR TITLE
Added ToStringStream

### PR DESCRIPTION
**Pull request by wdullaer**
This is similar to `ToArrayStream` and `ToObjectStream`: it buffers the events into a string.
It'll try to do the right thing based on the type of the events passed in.

I find it useful when testing code that streams into an expressjs response.

### CODE REVIEW

**Comment by wdullaer** [comment]
Hey,

Are you guys interested in this change?

**Comment by pgherveou** [comment]
Sorry didnt have time to review it. 
Will have a quick look now

**Comment by pgherveou** [comment]
so quick question 
I am not sure its super useful to add that in a generic toolbox like stream-util
wouldnt this be enough?
```js
toArray().then(arr => arr.join(''))
```
or 
```js
toArray().then(arr => arr.map(JSON.stringify).join(''))
```

File: lib/index.js [hunk 1]
```diff
@@ -130,6 +130,42 @@ class ToObjectStream extends Transform {
...
+    this._deferred = defer()
+  }
+
+  _transform (chunk: any, encoding: string, next: () => void) {
+    encoding = encoding === 'buffer' ? undefined : encoding
+    let str
+    if (Buffer.isBuffer(chunk)) str = chunk.toString(encoding)
+    else if (typeof chunk === 'string') str = chunk
+    else str = JSON.stringify(chunk)
+    this._buffer += str
```
Comment (pgherveou) [inline lib/index.js]: I think it's slightly more efficient to use an array buffer and in the _flush method use _buffer.join('')
https://jsperf.com/array-join-vs-string-connect

Comment (wdullaer) [inline lib/index.js]: That's easy enough to fix :-)

**Comment by wdullaer** [comment]
@pgherveou Using `toArray().join('')` would actually work fine.
I guess I overlooked the obvious while working on this.

Let me know what you'd like to do with it.

In other news: I have a bugfix for the error handling in `mapSync()`
I'm also thinking of making some utility that propagates errors through in a pipe if it's not handled, rather than just throwing. (Inspired by this: nodejs/node/#1521 ). Is that something you feel belongs here, or I put it somewhere separate?

**Comment by pgherveou** [comment]
lets close this PR then and start new ones for bugfixes and better error handling 
thanks for all your time and contributions, really appreciate it

**Comment by pgherveou** [comment]
did you have a look at this method 
https://github.com/jogabo/stream-util/blob/master/lib/index.js#L448
to handle errors in a a stream pipeline?

**Comment by wdullaer** [comment]
I haven't, I'll experiment with that today.

I'm happy to help out here. I'm getting a lot of mileage out of this and just try to fill the gaps I'm seeing (which might or might not be relevant)

**Pull request closed without merging**